AuthenticatorSyncShared/Tests/SharedKeychainRepositoryTests.swift

```diff
@@ -0,0 +1,108 @@
+import CryptoKit
+import Foundation
+import XCTest
+
+@testable import AuthenticatorSyncShared
+
+final class SharedKeychainRepositoryTests: XCTestCase {
```

Comment:
🤔 Could we add a base class for `XCTestCase` and get some of [BitwardenTestCase](https://github.com/bitwarden/ios/blob/main/GlobalTestHelpers/Support/BitwardenTestCase.swift) methods into it? Like `assertAsyncThrows` which are really handy to use. Additionally, I'd like to have a base class for tests in case we need to do changes affecting the entire test suite.

Reply:
Sounds good. I was trying to make sure we didn't duplicate too much, but totally understand wanting to have one base class to work from 👍 `assertAsyncThrows` will definitely be handy to have. I'll add in the next commit.

Reply:
Added this in the latest commit. I took out the UI-related stuff (partially because we don't have UI for this in the package, but also because that had some BitwardenShared dependencies as well).